scripts/Migration/XML_Conversion/destwork.sh

```diff
@@ -23,19 +23,25 @@ L_RRDPATH=/opt/librenms/rrd/
 ADDHOST=/opt/librenms/addhost.php
 # Enter your unique SNMP String
 SNMPSTRING=cisconetwork
+# Enter SNMP version of all clients in nodelist text file
+SNMPVERSION=v2c
 # Enter path to nodelist text file
 NODELIST=/tmp/nodelist.txt
+# Enter user and group of LibreNMS installation
+L_USRGRP=librenms
 
 # Loop enters RRD directory and then each folder based on contents of node list text file
-while read line 
+while read line
 	# Enter the directory
 	do cd $L_RRDPATH"${line%/*}"
 		# Convert from XML back to RRD
 		for f in *.xml; do rrdtool restore ${f} `echo ${f} | cut -f1 -d .`.rrd; done;
 		# Remove leftover XML files
 		rm *.xml;
+		# Change ownership to LibreNMS user and group
+		chown -R $L_USRGRP:$L_USRGRP .;
```

Comment:
Would that not introduce a race condition? Once addhost.php is called, if the poller runs before the ownership has been changed, it will not be able to rename or write to the files.